refinery/data_set_manager/tests.py

```diff
@@ -1777,34 +1782,176 @@ def test_prepare(self):
         )
 
 
-class AnnotatedNodeExplosionTestCase(TestCase):
+@contextlib.contextmanager
+def temporary_directory(*args, **kwargs):
+    d = tempfile.mkdtemp(*args, **kwargs)
+    try:
+        yield d
+    finally:
+        shutil.rmtree(d)
+
+
+class IsaTabParserTests(TestCase):
     def setUp(self):
         self.username = 'coffee_lover'
         self.password = 'coffeecoffee'
         self.user = User.objects.create_user(
-            self.username,
-            '',
-            self.password
+            self.username, '', self.password
         )
 
-    def test_metabolights_isatab_wont_import_with_rollback_a(self):
-        parse_isatab(
-            self.user.username,
-            True,
-            "refinery/data_set_manager/test-data/MTBLS1.zip"
-        )
+    def failed_isatab_assertions(self):
         self.assertEqual(DataSet.objects.count(), 0)
         self.assertEqual(AnnotatedNode.objects.count(), 0)
         self.assertEqual(Node.objects.count(), 0)
         self.assertEqual(FileStoreItem.objects.count(), 0)
+        self.assertEqual(Investigation.objects.count(), 0)
+
+    def parse(self, dir_name):
+        parent = os.path.dirname(os.path.abspath(__file__))
+        file_source_translator = generate_file_source_translator(
+            username=self.user.username
+        )
+        dir = os.path.join(parent, 'test-data', dir_name)
+        return IsaTabParser(
+            file_source_translator=file_source_translator
+        ).run(dir)
+
+    def test_empty(self):
+        with temporary_directory() as tmp:
+            with self.assertRaises(ParserException):
+                self.parse(tmp)
+
+    def test_minimal(self):
+        investigation = self.parse('minimal')
+
+        studies = investigation.study_set.all()
+        self.assertEqual(len(studies), 1)
+
+        assays = studies[0].assay_set.all()
+        self.assertEqual(len(assays), 1)
+
+    def test_mising_investigation(self):
+        with self.assertRaises(ParserException):
+            self.parse('missing-investigation')
+
+    def test_mising_study(self):
+        with self.assertRaises(IOError):
```

Comment:
We report all errors to the user, and not just ParserExceptions, right? Even so, might check that the information returned by this is enough to act on.

Reply:
@mccalluc Yep we'll catch anything unexpected. This case will show in the ui like so:
![screen shot 2017-10-10 at 11 37 04 am](https://user-images.githubusercontent.com/5629547/31395702-6190a55e-adaf-11e7-9423-6d8562a8ac9d.png)